lib/magma/attribute.rb

```diff
@@ -15,45 +34,56 @@ class Attribute
       :validation
     ]
 
-    attr_reader :name, :loader, :validation, :format_hint, :unique, :index, :restricted, :link_model_name, :description
+    attr_reader :loader
 
     class << self
       def options
         [:description, :display_name, :hidden, :read_only, :unique, :index, :validation,
 :format_hint, :loader, :link_model_name, :restricted]
       end
 
-      def set_attribute(name, model, options, attribute_class)
-        attribute_class.new(name, model, options)
-      end
-
       def attribute_type
         @attribute_type ||= name.match("Magma::(.*)Attribute")[1].underscore
       end
     end
 
-    def initialize(name, model, opts)
-      @name = name
-      @model = model
-      set_options(opts)
+    def initialize(opts = {})
+      # Some Ipi models set the group option but it doesn't seem to be used anywhere
+      opts.delete(:group)
```

Comment:
Like the comment says, some Ipi models set the `group` option but I didn't find any uses of it in magma. Do we need to support it?

Reply:
We're interested in using this more in the future to help organize the attribute space; but we haven't really developed the notion fully so it can be left aside for the moment.